src/StoreApi/Schemas/ShippingAddressSchema.php

```diff
@@ -36,6 +36,12 @@ class ShippingAddressSchema extends AbstractAddressSchema {
 	 */
 	public function get_item_response( $address ) {
 		if ( ( $address instanceof \WC_Customer || $address instanceof \WC_Order ) ) {
+			if ( is_callable( [ $address, 'get_shipping_phone' ] ) ) {
+				$shipping_phone = $address->get_shipping_phone();
+			} else {
+				$shipping_phone = $address->get_meta( $address instanceof \WC_Customer ? 'shipping_phone' : '_shipping_phone', true );
+			}
```

Comment:
So one potential problem I see here (and elsewhere similar retrieval logic is found), is that should we release Checkout I2 in the feature plugin (non-experimental) before `get_shipping_phone` is available in core, and then later the store updates to a version of WooCommerce core that _does_ have `get_shipping_phone` is there risk here that the existing shipping phone stored in the `shipping_phone` or `_shipping_phone` meta fields would not be referenced leading to displaying/saving an empty value because `get_shipping_phone` doesn't return any value? 

As far as I can tell there's no migration routine for handling this sort of scenario and if we leave this in place I wonder if we should have something like this logic instead:

```suggestion
			$shipping_phone = $address->get_meta( $address instanceof \WC_Customer ? 'shipping_phone' : '_shipping_phone', true );
			if ( empty( $shipping_phone ) && is_callable( [ $address, 'get_shipping_phone' ] ) ) {
				$shipping_phone = $address->get_shipping_phone();
			}
```

It does result in two db queries but reduces the risk of losing information?

_This comment applies similarly in other places where there is logic conditionally retrieving the shipping phone value._

Reply:
I think the way I have it round is better for two reasons:

1. We're defaulting to the core method when it's available. _It looks like this will be available in WC 5.6_ 
2. The fallback logic I used is using the correct meta key which will match the core implementation. That's why one is prefixed and one is not.

Customer data store in core: https://github.com/woocommerce/woocommerce/blob/trunk/includes/data-stores/class-wc-customer-data-store.php#L305

Order data store: https://github.com/woocommerce/woocommerce/blob/trunk/includes/data-stores/class-wc-order-data-store-cpt.php#L137

Because we use the same meta key, there will be no data loss.

Reply:
Ahh, that makes sense, I missed the piece about the two implementations sharing the same meta key 👍🏻